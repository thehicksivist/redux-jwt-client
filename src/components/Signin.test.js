import React from 'react'
import renderer from 'react-test-renderer'
import SignIn from './SignIn'

describe('Bookmark snapshot', () => {
    it('Should render as expected with login error', () => {
        const tree = renderer.create(
            // <SignIn loginError={store.getState().loginError} handleSignIn={this.handleSignIn} />
            // the above is what is defined is in our app.js, so we need to pass in the expected fields 
            // in the test (below)
            <SignIn loginError="login error" handleSignIn={() => {}} />
        )
        // tree is the above mock-up 
        expect(tree.toJSON()).toMatchSnapshot()
    })
})

describe('Bookmark snapshot', () => {
    it('Should render as expected without login error', () => {
        const tree = renderer.create(
            // <SignIn loginError={store.getState().loginError} handleSignIn={this.handleSignIn} />
            // the above is what is defined is in our app.js, so we need to pass in the expected fields 
            // in the test (below), the {} are used to write js
            <SignIn loginError={null} handleSignIn={() => {}} />
        )
        // tree is the above mock-up 
        expect(tree.toJSON()).toMatchSnapshot()
    })
})