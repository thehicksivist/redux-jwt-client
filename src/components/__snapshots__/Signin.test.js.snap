// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark snapshot Should render as expected 1`] = `
<li>
  My Bookmark
   (
  <a
    href="http://www.mybookmark.com"
  >
    Visit
  </a>
  )
  <button
    onClick={[Function]}
  >
    Delete!
  </button>
</li>
`;

exports[`Bookmark snapshot Should render as expected with login error 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
    login error
  </p>
  <label>
    Email: 
    <input
      name="email"
      type="email"
    />
  </label>
  <br />
  <label>
    Password: 
    <input
      name="password"
      type="password"
    />
  </label>
  <br />
  <button
    type="submit"
  >
    Login
  </button>
</form>
`;

exports[`Bookmark snapshot Should render as expected without login error 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Email: 
    <input
      name="email"
      type="email"
    />
  </label>
  <br />
  <label>
    Password: 
    <input
      name="password"
      type="password"
    />
  </label>
  <br />
  <button
    type="submit"
  >
    Login
  </button>
</form>
`;
